### PR TITLE
feat: mobile card detail screen

### PR DESCRIPTION
**Pull request by hyj1204**
### Feature Preview

**Card detail page**
This PR is only for the card detail page, the card style in board view will be added in the future PR.
Appbar: 
- Duplicate & Delete card. After Duplicate/Delete actions, return to the viewPage(2 pages pop up) and show a toast.

https://github.com/AppFlowy-IO/AppFlowy/assets/14248245/247bc6f5-c8b5-4def-bc0a-8f49b2acc195

Body: 
The other types of properties will be added(fit to mobile) in the future PR
- Edit card title
- Show properties: Text, Numbers, Last modified, Created time, Datetime, Checkbox
- Edit properties:  Text, ~~Numbers~~, Last modified, Created time, Datetime, Checkbox

https://github.com/AppFlowy-IO/AppFlowy/assets/14248245/bbadec96-12c4-40f6-8a09-f35d6d9888f0

- Add properties: Text, ~~Numbers~~, Last modified, Created time, Datetime, Checkbox

https://github.com/AppFlowy-IO/AppFlowy/assets/14248245/1cbdef89-8fa2-401a-9e66-514dbc19f243

- Edit property value: Text, Numbers, Datetime, Checkbox

https://github.com/AppFlowy-IO/AppFlowy/assets/14248245/35cae49f-b350-4b67-ab32-8dffdf469a69

- Delete properties
- Reorder properties: Long pressed the property to move the order of it
- Show/Hidden fields: Use the same widget as the Desktop

https://github.com/AppFlowy-IO/AppFlowy/assets/14248245/c652c3d7-bcfc-4037-bbe1-004a3aecb13a

**Found Issue:**

The card detail page cannot be rendered when switching back to itself through the navigation bar: I plan to fix this problem while working on the board view page.

---

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/mobile_card_detail_screen.dart [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  final scrollController = ScrollController();
+
+  @override
+  void dispose() {
+    scrollController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/mobile_card_detail_screen.dart:50]: Split the tree into smaller widgets.

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/mobile_create_row_field_screen.dart [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+
+class MobileCreateRowFieldScreen extends StatefulWidget {
+  final String viewId;
+  final TypeOptionPB typeOption;
+  static const routeName = '/MobileCreateRowFieldScreen';
+  static const argViewId = 'viewId';
+  static const argTypeOption = 'typeOption';
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/mobile_create_row_field_screen.dart:14]: Place the properties after the constructor and maintain the same style, like this.

Screenshot 2023-11-09 at 11 39 47

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/widgets/mobile_field_name_text_field.dart [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+class MobileFieldNameTextField extends StatefulWidget {
+  const MobileFieldNameTextField({
+    this.text,
+    super.key,
+    this.textEditingController,
+  });
+
+  final String? text;
+  final TextEditingController? textEditingController;
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/widgets/mobile_field_name_text_field.dart:13]: Is it unused?

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/widgets/mobile_field_name_text_field.dart [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    super.initState();
+    controller = widget.textEditingController ?? TextEditingController();
+    if (widget.text != null) {
+      controller.text = widget.text!;
+    }
+  }
+
+  @override
+  void dispose() {
+    controller.dispose();
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/widgets/mobile_field_name_text_field.dart:33]: If the controller is not created internally, then you shouldn't dispose it. The entity that creates it should be responsible for disposing it.

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/widgets/mobile_row_property_list.dart [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flowy_infra_ui/flowy_infra_ui.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:go_router/go_router.dart';
+
+/// Display the row properties in a list. Only use this widget in the
+/// [MobileCardDetailScreen].
+class MobileRowPropertyList extends StatelessWidget {
+  final String viewId;
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/widgets/mobile_row_property_list.dart:24]: Keep the same structure here.

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/widgets/mobile_row_property_list.dart [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+  final GridCellBuilder cellBuilder;
+  const MobileRowPropertyList({
+    required this.viewId,
+    required this.cellBuilder,
+    Key? key,
+  }) : super(key: key);
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/widgets/mobile_row_property_list.dart:30]: super.key

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/widgets/mobile_row_property_list.dart [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+            padding: EdgeInsets.zero,
+          ),
+          // naivgator to field editor
+          onPressed: () => context.push(
+            CardPropertyEditScreen.routeName,
+            extra: {
+              CardPropertyEditScreen.argCellContext: widget.cellContext,
+              CardPropertyEditScreen.argRowDetailBloc:
+                  context.read<RowDetailBloc>(),
+            },
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/widgets/mobile_row_property_list.dart:146]: I don't think it's a good way to pass the block and context through the router. The parameters should be basic types, like string and int, and then you can create the necessary block and other components.

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/widgets/mobile_field_name_text_field.dart (hunk 3 above)
Comment (hyj1204) [reply frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/widgets/mobile_field_name_text_field.dart:13]: I deleted `textEditingController` here, and used a TextEditingController internally (dispose it)

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/widgets/mobile_row_property_list.dart (hunk 7 above)
Comment (hyj1204) [reply frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/widgets/mobile_row_property_list.dart:146]: I agree. I plan to refactor here while working on the board view. Currently, this part code still depends on the shared code with the desktop. I probably will refract the upper layers of code to fit the mobile needs.

**Pull request closed without merging**